test/unit/org/apache/cassandra/tools/nodetool/ClearSnapshotTest.java

```diff
@@ -52,60 +71,142 @@ public static void teardown() throws IOException
+        tool = invokeNodetool("clearsnapshot", "--all");
         tool.assertOnCleanExit();
     }
 
     @Test
     public void testClearSnapshot_AllAndName()
     {
-        ToolRunner.ToolResult tool = ToolRunner.invokeNodetool("clearsnapshot", "-t", "some-name", "--all");
+        ToolRunner.ToolResult tool = invokeNodetool("clearsnapshot", "-t", "some-name", "--all");
         assertThat(tool.getExitCode()).isEqualTo(2);
         assertThat(tool.getCleanedStderr()).contains("Specify only one of snapshot name or --all");
     }
 
     @Test
     public void testClearSnapshot_RemoveByName()
     {
-        ToolRunner.ToolResult tool = ToolRunner.invokeNodetool("snapshot","-t","some-name");
+        ToolRunner.ToolResult tool = invokeNodetool("snapshot","-t","some-name");
         tool.assertOnCleanExit();
         assertThat(tool.getStdout()).isNotEmpty();
         
-        Map<String, TabularData> snapshots_before = probe.getSnapshotDetails();
+        Map<String, TabularData> snapshots_before = probe.getSnapshotDetails(emptyMap());
         assertThat(snapshots_before).containsKey("some-name");
 
-        tool = ToolRunner.invokeNodetool("clearsnapshot","-t","some-name");
+        tool = invokeNodetool("clearsnapshot","-t","some-name");
         tool.assertOnCleanExit();
         assertThat(tool.getStdout()).isNotEmpty();
         
-        Map<String, TabularData> snapshots_after = probe.getSnapshotDetails();
+        Map<String, TabularData> snapshots_after = probe.getSnapshotDetails(emptyMap());
         assertThat(snapshots_after).doesNotContainKey("some-name");
     }
 
     @Test
     public void testClearSnapshot_RemoveMultiple()
     {
-        ToolRunner.ToolResult tool = ToolRunner.invokeNodetool("snapshot","-t","some-name");
+        ToolRunner.ToolResult tool = invokeNodetool("snapshot","-t","some-name");
         tool.assertOnCleanExit();
         assertThat(tool.getStdout()).isNotEmpty();
 
-        tool = ToolRunner.invokeNodetool("snapshot","-t","some-other-name");
+        tool = invokeNodetool("snapshot","-t","some-other-name");
         tool.assertOnCleanExit();
         assertThat(tool.getStdout()).isNotEmpty();
 
-        Map<String, TabularData> snapshots_before = probe.getSnapshotDetails();
+        Map<String, TabularData> snapshots_before = probe.getSnapshotDetails(emptyMap());
         assertThat(snapshots_before).hasSize(2);
 
-        tool = ToolRunner.invokeNodetool("clearsnapshot","--all");
+        tool = invokeNodetool("clearsnapshot","--all");
         tool.assertOnCleanExit();
         assertThat(tool.getStdout()).isNotEmpty();
         
-        Map<String, TabularData> snapshots_after = probe.getSnapshotDetails();
+        Map<String, TabularData> snapshots_after = probe.getSnapshotDetails(emptyMap());
         assertThat(snapshots_after).isEmpty();
     }
+
+    @Test
+    public void testClearSnapshotWithOlderThanFlag() throws Throwable
```

Comment:
Are we testing `olderThanTimestamp` ?

Reply:
We are, at the bottom of that test method. I added a dedicated test method for it so we do not mix two flags together.